src/main/java/org/opensearch/ad/rest/RestExecuteAnomalyDetectorAction.java

```diff
@@ -19,73 +19,104 @@
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
 
 import org.apache.commons.lang.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.opensearch.ad.AnomalyDetectorPlugin;
+import org.opensearch.action.ActionListener;
+import org.opensearch.ad.AnomalyDetectorExtension;
 import org.opensearch.ad.constant.CommonErrorMessages;
 import org.opensearch.ad.model.AnomalyDetectorExecutionInput;
+import org.opensearch.ad.settings.AnomalyDetectorSettings;
 import org.opensearch.ad.settings.EnabledSetting;
 import org.opensearch.ad.transport.AnomalyResultAction;
 import org.opensearch.ad.transport.AnomalyResultRequest;
-import org.opensearch.client.node.NodeClient;
-import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.ad.transport.AnomalyResultResponse;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.xcontent.ToXContent;
 import org.opensearch.core.xcontent.XContentParser;
-import org.opensearch.rest.BaseRestHandler;
-import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
 import org.opensearch.rest.RestRequest;
 import org.opensearch.rest.RestStatus;
-import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.sdk.SDKClient.SDKRestClient;
+import org.opensearch.sdk.rest.BaseExtensionRestHandler;
+import org.opensearch.sdk.rest.ReplacedRouteHandler;
 
 import com.google.common.collect.ImmutableList;
 
 /**
  * This class consists of the REST handler to handle request to detect data.
  */
-public class RestExecuteAnomalyDetectorAction extends BaseRestHandler {
+public class RestExecuteAnomalyDetectorAction extends BaseExtensionRestHandler {
 
     public static final String DETECT_DATA_ACTION = "execute_anomaly_detector";
     // TODO: apply timeout config
     private volatile TimeValue requestTimeout;
-
+    private SDKRestClient sdkRestClient;
+    private Settings settings;
     private final Logger logger = LogManager.getLogger(RestExecuteAnomalyDetectorAction.class);
 
-    public RestExecuteAnomalyDetectorAction(Settings settings, ClusterService clusterService) {
-        this.requestTimeout = REQUEST_TIMEOUT.get(settings);
-        clusterService.getClusterSettings().addSettingsUpdateConsumer(REQUEST_TIMEOUT, it -> requestTimeout = it);
+    public RestExecuteAnomalyDetectorAction(ExtensionsRunner extensionsRunner, SDKRestClient sdkRestClient) {
+        this.requestTimeout = REQUEST_TIMEOUT.get(extensionsRunner.getEnvironmentSettings());
+        this.sdkRestClient = sdkRestClient;
+        extensionsRunner.getSdkClusterService().getClusterSettings().addSettingsUpdateConsumer(REQUEST_TIMEOUT, it -> requestTimeout = it);
+        this.settings = extensionsRunner.getEnvironmentSettings();
```

Comment:
You can define this at the top and use it for 
`this.requestTimeout = REQUEST_TIMEOUT.get(settings);`

Reply:
ACK